packages/toolkit/src/query/tests/utils.test.ts

```diff
@@ -99,6 +101,10 @@ describe('joinUrls', () => {
     expect(joinUrls('https://example.com/api/', '/banana/')).toBe(
       'https://example.com/api/banana/'
     )
+
+    expect(joinUrls('https://example.com/api/', '?foo=bar')).toBe(
+      'https://example.com/api?foo=bar'
+    )
```

Comment:
I'm too late to the party :(

I feel like this scenario is a bug. There are servers that expect the slash and will lead you to a completely different page without it - completely independently of a search param being present.

It's fine to *not add* the slash - but actively removing ti is dangerous.